tests/unit/test_row_noise.py

```diff
@@ -294,22 +298,38 @@ def test_guardian_duplication() -> None:
     census = Dataset(DATASET_SCHEMAS.census, dummy_data, 0)
     NOISE_TYPES.duplicate_with_guardian(census, config)
     noised = census.data
-    # We know the following since every dependent is duplicated:
-    #  - Simulant ids 0, 1, 2, 3, and-5 will all be duplicated
-    #  - Simulant ids 5-9 are guardians. The only overlap is simulant id 5,
-    #    who is both a dependent and a guardian
-    #  - Simulant id 0 and 3 have two guardians, 8 and 9.
 
-    # Check that the correct rows were duplicated. Duplicated returns all instances of True after
-    # the first instance
     duplicated = noised.loc[noised["simulant_id"].duplicated()]
-    guardians = dummy_data.loc[dummy_data["simulant_id"].isin(dummy_data["guardian_1"])]
-    assert len(noised) == len(dummy_data) + len(duplicated)
-    assert set(duplicated["simulant_id"].tolist()) == set(["0", "1", "2", "3", "5"])
-    # Only duplicate a depedent one time
+
+    if duplication_probability == 1.0:
+        # We know the following since every dependent is duplicated
+        # in the case of 100% duplication:
+        #  - Simulant ids 0, 1, 2, 3, and-5 will all be duplicated
+        #  - Simulant ids 5-9 are guardians. The only overlap is simulant id 5,
+        #    who is both a dependent and a guardian
+        #  - Simulant id 0 and 3 have two guardians, 8 and 9.
+
+        # Check that the correct rows were duplicated. Duplicated returns all
+        # instances of True after the first instance
+        assert len(noised) == len(dummy_data) + len(duplicated)
+        assert set(duplicated["simulant_id"].tolist()) == set(["0", "1", "2", "3", "5"])
+    else: # non-1 probability
+        has_guardian = dummy_data['guardian_1'].notna()
+        not_in_military = dummy_data['housing_type'] != 'Military'
+        num_eligible_for_duplication = sum(has_guardian & not_in_military)
+
+        fuzzy_checker.fuzzy_assert_proportion(
+            name="test_do_not_respond",
+            observed_numerator=len(duplicated),
+            observed_denominator=num_eligible_for_duplication,
+            target_proportion=duplication_probability,
+            name_additional=f"noised_data",
+        )
+    # Only duplicate a dependent one time
     assert noised["simulant_id"].value_counts().max() == 2
 
     # Check address information is copied in new rows
+    guardians = dummy_data.loc[dummy_data["simulant_id"].isin(dummy_data["guardian_1"])]
```

Comment:
This line was moved from where it was in the old test to right before it's actually used.